akka-remote/src/main/scala/akka/remote/artery/ArterySubsystem.scala

```diff
@@ -67,30 +67,27 @@ private[remote] class ArterySubsystem(_system: ExtendedActorSystem, _provider: R
 
     val association =
       if (cached ne null) cached
-      else {
-        val association = getAssociation(remoteAddress)
-        association.associate()
-        recipient.cachedAssociation = association
-        association
-      }
+      else associate(remoteAddress)
 
     association.send(message, senderOption, recipient)
   }
 
-  private def getAssociation(remoteAddress: Address): Association = {
+  private def associate(remoteAddress: Address): Association = {
     val current = associations.get(remoteAddress)
     if (current ne null) current
```

Comment:
we must also, perhaps later, think about how to handle races when removing associations